app/domains/fluggastrechte/formular/services/validation.ts

```diff
@@ -0,0 +1,224 @@
+      ctx.addIssue({
+        code: z.ZodIssueCode.custom,
+        message: "arrivalThreeHoursLessThanDeparture",
+        path: ["andereErsatzverbindungAnkunftsZeit"],
+        fatal: true,
+      });
+
+      return z.NEVER;
+    }
+  });
+}
+
+export function validateAnotherFlightPage(
+  baseSchema: MultiFieldsValidationBaseSchema,
+) {
+  return baseSchema.superRefine((data, ctx) => {
+    const departureDateTime = convertToTimestamp(
+      data.direktAbflugsDatum,
+      data.direktAbflugsZeit,
+    );
+
+    const arrivalDateTime = convertToTimestamp(
+      data.ersatzFlugAnkunftsDatum,
+      data.ersatzFlugAnkunftsZeit,
+    );
+
+    if (departureDateTime > arrivalDateTime) {
+      ctx.addIssue({
+        code: z.ZodIssueCode.custom,
+        message: "departureAfterArrival",
+        path: ["ersatzFlugAnkunftsDatum"],
+        fatal: true,
+      });
+
+      // add new issue to invalidate this field as well
+      ctx.addIssue({
+        code: z.ZodIssueCode.custom,
+        message: "departureAfterArrival",
+        path: ["ersatzFlugAnkunftsZeit"],
+        fatal: true,
+      });
+
+      return z.NEVER;
+    }
+
+    if (
+      isStartTimestampLessThanThreeHours(departureDateTime, arrivalDateTime) &&
+      data.bereich === "verspaetet"
+    ) {
+      ctx.addIssue({
+        code: z.ZodIssueCode.custom,
+        message: "arrivalThreeHoursLessThanDeparture",
+        path: ["ersatzFlugAnkunftsDatum"],
+        fatal: true,
+      });
+
+      // add new issue to invalidate this field as well
+      ctx.addIssue({
+        code: z.ZodIssueCode.custom,
+        message: "arrivalThreeHoursLessThanDeparture",
+        path: ["ersatzFlugAnkunftsZeit"],
+        fatal: true,
+      });
+
+      return z.NEVER;
+    }
+  });
+}
+
+export function validateSameFlightPage(
+  baseSchema: MultiFieldsValidationBaseSchema,
+) {
+  return baseSchema.superRefine((data, ctx) => {
+    const departureDateTime = convertToTimestamp(
+      data.direktAbflugsDatum,
+      data.direktAbflugsZeit,
+    );
+
+    const arrivalDateTime = convertToTimestamp(
+      data.tatsaechlicherAnkunftsDatum,
+      data.tatsaechlicherAnkunftsZeit,
+    );
+
+    if (departureDateTime > arrivalDateTime) {
+      ctx.addIssue({
+        code: z.ZodIssueCode.custom,
+        message: "departureAfterArrival",
+        path: ["tatsaechlicherAnkunftsDatum"],
+        fatal: true,
+      });
+
+      // add new issue to invalidate this field as well
+      ctx.addIssue({
+        code: z.ZodIssueCode.custom,
+        message: "departureAfterArrival",
+        path: ["tatsaechlicherAnkunftsZeit"],
+        fatal: true,
+      });
+
+      return z.NEVER;
+    }
+
+    if (
+      isStartTimestampLessThanThreeHours(departureDateTime, arrivalDateTime)
+    ) {
+      ctx.addIssue({
+        code: z.ZodIssueCode.custom,
+        message: "arrivalThreeHoursLessThanDeparture",
+        path: ["tatsaechlicherAnkunftsDatum"],
```

Comment:
Since this is an array, is it possible to combine it with the AnkunftsDatum and AnkunftsZeit so we won't need to call `ctx.addIssue()` again, or it has to be separated?

```suggestion
        path: ["tatsaechlicherAnkunftsDatum", "tatsaechlicherAnkunftsZeit"],
```

Reply:
if we can put it together, then we can reduce all `ctx.addIssue()` in this file

Reply:
I tried, but it doesn't work. The remix hook `useField` don't recognise arrays on the fieldName parameter, only a normal string.